kissfft.pc.in

```diff
@@ -1,9 +1,10 @@
 libdir=@CMAKE_INSTALL_FULL_LIBDIR@
 includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
+pkgincludedir=@CMAKE_INSTALL_FULL_PKGINCLUDEDIR@
 
 Name: kissfft
 Description: A Fast Fourier Transform (FFT) library that tries to Keep it Simple, Stupid
 Version: @kissfft_VERSION@
 
-Libs: -L${libdir} -l@KISSFFT_OUTPUT_NAME@
-Cflags: -I${includedir} @PKG_KISSFFT_DEFS@
\ No newline at end of file
+Libs: @PKG_OPENMP@ -L${libdir} -l@KISSFFT_OUTPUT_NAME@
+Cflags: -I${includedir} -I${pkgincludedir} @PKG_KISSFFT_DEFS@
```

Comment:
This is the same as before but with a different variable name and even less idiomatic.

Reply:
To be honest, I dont understand what is wrong here. If I specify `${includedir}/kissfft` here, I will have to suppress the ability to override `PKGINCLUDEDIR`. And I have to edit both `Makefile` and `CMakeLists.txt`...

Reply:
There should be no need to override `pkgincludedir`, people can already override the includedir.

The most idiomatic would be something like:

```patch
diff --git a/CMakeLists.txt b/CMakeLists.txt
index bb72a0f..f4cf632 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -106,14 +106,10 @@ if(CMAKE_SYSTEM_NAME MATCHES "^(Linux|kFreeBSD|GNU)$" AND NOT CMAKE_CROSSCOMPILI
 endif()
 
 #
-# Declare CMAKE_INSTALL_{FULL_,}PKGINCLUDEDIR for kissfft include path
+# Declare PKGINCLUDEDIR for kissfft include path
 #
 
-if (NOT DEFINED CMAKE_INSTALL_PKGINCLUDEDIR)
-    set(CMAKE_INSTALL_PKGINCLUDEDIR "${CMAKE_INSTALL_INCLUDEDIR}/kissfft")
-    set(CMAKE_INSTALL_FULL_PKGINCLUDEDIR "${CMAKE_INSTALL_FULL_INCLUDEDIR}/kissfft")
-    message(STATUS "CMAKE_INSTALL_PKGINCLUDEDIR is ${CMAKE_INSTALL_PKGINCLUDEDIR}")
-endif()
+set(PKGINCLUDEDIR "${CMAKE_INSTALL_INCLUDEDIR}/kissfft")
 
 #
 # Declare kissfft library ( libkissfft.a / libkissfft-${KISSFFT_DATATYPE}.so.${MAKEFILE_EXTRACTED_VERSION} )
@@ -128,7 +124,7 @@ add_library(kissfft
 
 target_include_directories(kissfft PUBLIC
     $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}>
-    $<INSTALL_INTERFACE:${CMAKE_INSTALL_PKGINCLUDEDIR}>)
+    $<INSTALL_INTERFACE:${PKGINCLUDEDIR}>)
 
 #
 # Set compile definitions based on datatype and additional support flags
@@ -286,7 +282,7 @@ install(FILES kiss_fft.h
               kiss_fftnd.h
               kiss_fftndr.h
               kiss_fftr.h
-        DESTINATION "${CMAKE_INSTALL_PKGINCLUDEDIR}")
+        DESTINATION "${PKGINCLUDEDIR}")
 
 set(KISSFFT_INSTALL_CMAKE "${CMAKE_INSTALL_LIBDIR}/cmake/${PROJECT_NAME}"
     CACHE FILEPATH "Install destination of kissfft cmake modules")
@@ -307,6 +303,11 @@ foreach(_def ${KISSFFT_COMPILE_DEFINITIONS})
     set(PKG_KISSFFT_DEFS "${PKG_KISSFFT_DEFS} -D${_def}")
 endforeach()
 if (KISSFFT_PKGCONFIG)
+    set(pkgincludedir_for_pkg_config "\${includedir}/kissfft")
+    set(prefix_for_pkg_config "${CMAKE_INSTALL_PREFIX}")
+    join_paths(libdir_for_pkg_config "\${prefix}" "${CMAKE_INSTALL_LIBDIR}")
+    join_paths(includedir_for_pkg_config "\${prefix}" "${CMAKE_INSTALL_INCLUDEDIR}")
+
     if(KISSFFT_DATATYPE MATCHES "^simd$")
        list(APPEND KISSFFT_COMPILE_DEFINITIONS USE_SIMD)
        if (NOT MSVC)
diff --git a/kissfft.pc.in b/kissfft.pc.in
index 3f130ad..322147a 100644
--- a/kissfft.pc.in
+++ b/kissfft.pc.in
@@ -1,10 +1,10 @@
-libdir=@CMAKE_INSTALL_FULL_LIBDIR@
-includedir=@CMAKE_INSTALL_FULL_INCLUDEDIR@
-pkgincludedir=@CMAKE_INSTALL_FULL_PKGINCLUDEDIR@
+prefix=@prefix_for_pkg_config@
+libdir=@libdir_for_pkg_config@
+includedir=@includedir_for_pkg_config@
 
 Name: kissfft
 Description: A Fast Fourier Transform (FFT) library that tries to Keep it Simple, Stupid
 Version: @kissfft_VERSION@
 
 Libs: @PKG_OPENMP@ -L${libdir} -l@KISSFFT_OUTPUT_NAME@
-Cflags: -I${includedir} -I${pkgincludedir} @PKG_KISSFFT_DEFS@
+Cflags: -I@pkgincludedir_for_pkg_config@ @PKG_KISSFFT_DEFS@
```

but you would need [`join_paths`](https://github.com/jtojnar/cmake-snips/blob/3cf4ad9c4901a5d116adcf2773cad7097a3580c1/CMakeScripts/JoinPaths.cmake) macro for older versions of CMake.

Reply:
Thanks! Let me incorporate that diff, adjust Makefile counterpart & run the installation testsuite.

Reply:
Done